sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -2318,6 +2318,56 @@ object functions {
     new XxHash64(cols.map(_.expr))
   }
 
+  /**
+   * Returns null if the condition is true, and throws an exception otherwise.
+   *
+   * @group misc_funcs
+   * @since 3.1.0
+   */
+  def assert_true(c: Column): Column = withExpr {
+    new AssertTrue(c.expr)
+  }
+
+  /**
+   * Returns null if the condition is true; throws an exception with the error message otherwise.
+   *
+   * @group misc_funcs
+   * @since 3.1.0
+   */
+  def assert_true(c: Column, e: Column): Column = withExpr {
+    new AssertTrue(c.expr, e.expr)
+  }
+
+  /**
+   * Returns null if the condition is true; throws an exception with the error message otherwise.
+   *
+   * @group misc_funcs
+   * @since 3.1.0
+   */
+  def assert_true(c: Column, e: String): Column = withExpr {
```

Comment:
I would avoid exposing `String` overridden ones. See also https://github.com/apache/spark/blob/1b60ff5afea0637f74c5f064642225b35b13b069/sql/core/src/main/scala/org/apache/spark/sql/functions.scala#L58-L60

Reply:
I see - I can expose only Column inputs, and the user can wrap any literal strings.